config/read.go

```diff
@@ -760,7 +765,12 @@ func Read(logger *util.Logger, filename string) (Config, error) {
 			if err != nil {
 				return conf, err
 			}
-			config.SectionName = section.Name()
+
+			if config.GetDbName() == "" {
+				return conf, fmt.Errorf("No connection info found for section %s; see https://pganalyze.com/docs/collector/settings", sectionName)
```

Comment:
Is an error for the whole file really the best way to deal with this?

I agree that we shouldn't silent when there are invalid URLs, but I think generally our approach is that if one section is invalid (e.g. we can't connect to the server), we still handle the other ones. I'm not sure I see the need to make this more than a `PrintError` (though I don't think this is a problem per se, just find the behavior inconsistent).

Reply:
I think failing when reading the configuration file is the best time to fail, since someone is more likely to be interacting with the collector and likely to notice the error. I think something like "no connection info at all" is different than "could not connect to this server right now". But I don't feel strongly about this; I'll change this back to a PrintError.

Reply:
Yeah, I see your point - my main concern is that auto-generated config files might not have much interactivity, and one broken section shouldn't cause the whole file to fail (e.g. thinking someone didn't escape the password well, or something). I think having a PrintError is a clear improvement to what we have currently :)